### PR TITLE
Adjustments of loader-v4 (part 3)

### PR DESCRIPTION
**Pull request by Lichtso**
#### Problem
Continuation of #2750.

#### Summary of Changes
- Removes `load_and_finalize_program()` and `load_program()` in loader utils.
- Adds `instructions_to_load_program_of_loader_v4()` and `load_program_of_loader_v4()` to loader utils.
- Fixes incorrect program runtime environment in deployment verification check.
- Introduces minimum of 1 lamport to avoid program account from being deleted if rent is disabled.

### CODE REVIEW

File: runtime/src/loader_utils.rs [hunk 1]
```diff
@@ -65,63 +65,6 @@ pub fn create_program(bank: &Bank, loader_id: &Pubkey, name: &str) -> Pubkey {
...
-        bank_client
-            .send_and_confirm_message(&[payer_keypair, &program_keypair], message)
-            .unwrap();
-        offset += chunk_size as u32;
-    }
-    let instruction = loader_instruction::finalize(&program_keypair.pubkey(), loader_id);
-    (program_keypair, instruction)
-}
-
-pub fn load_program<T: Client>(
```
Comment (pgarg66) [inline runtime/src/loader_utils.rs:110]: Was this function unused?

Comment (Lichtso) [reply runtime/src/loader_utils.rs:110]: Yes they were for the loaders-v1 and v2, which we deprecated.

Comment (pgarg66) [reply runtime/src/loader_utils.rs:110]: So these are pure deletion of the code. The new functions are not related to these.
(trying to avoid comparing two independent functions).

Comment (Lichtso) [reply runtime/src/loader_utils.rs:110]: Yes, implementation / code wise they are unrelated. Conceptually they are related because these are their replacements but for the newest loader. However, loader-v4 is closest to loader-v3 so comparing the new functions against `load_upgradeable_program_and_advance_slot()` makes more sense.

File: runtime/src/loader_utils.rs [hunk 2]
```diff
@@ -313,6 +256,91 @@ pub fn set_upgrade_authority<T: Client>(
...
+    );
+    let signers: &[&[&Keypair]] = &[
+        &[payer_keypair, &program_keypair],
+        &[payer_keypair, &program_keypair, authority_keypair],
+        &[payer_keypair, authority_keypair],
+    ];
+    for (index, instruction) in instructions.into_iter().enumerate() {
+        let message = Message::new(&[instruction], Some(&payer_keypair.pubkey()));
+        bank_client
+            .send_and_confirm_message(signers[index.min(signers.len() - 1)], message)
```
Comment (pgarg66) [inline runtime/src/loader_utils.rs]: Some documentation will be helpful here, as to why we are doing `signers[index.min(signers.len() - 1)]`. I am able to understand it right now, but may not be obvious for the future maintainers.

File: programs/loader-v4/src/lib.rs [hunk 3]
```diff
@@ -279,7 +280,7 @@ pub fn process_instruction_deploy(
     };
     let executor = ProgramCacheEntry::new(
         &loader_v4::id(),
-        environments.program_runtime_v2.clone(),
+        environments.program_runtime_v1.clone(),
```
Comment (pgarg66) [inline programs/loader-v4/src/lib.rs:283]: Should we split this in its own PR?

**Comment by pgarg66** [comment]
The code looks good to me. Can we add some tests?

**Comment by Lichtso** [comment]
> Can we add some tests?

The tests will come in #2796. However we can't merge that until #2182, which depends on [SIMD-0162](https://github.com/solana-foundation/solana-improvement-documents/pull/162).

File: programs/loader-v4/src/lib.rs (hunk 3 above)
Comment (Lichtso) [reply programs/loader-v4/src/lib.rs:283]: This should have been in #2750 already but I missed it. Only noticed as I was working on the tests in #2796.

File: runtime/src/loader_utils.rs (hunk 2 above)
Comment (Lichtso) [reply runtime/src/loader_utils.rs]: I reimplemented it using iterators instead of indices, hopefully that makes it self explanatory.

**Review by pgarg66 — APPROVED** [review]
